CMakeLists.txt

```diff
@@ -86,6 +86,12 @@ endif()
 install(DIRECTORY ${PROJECT_BINARY_DIR}/include/ DESTINATION include
         FILES_MATCHING PATTERN "*.hpp")
 
+if (NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE "Debug" CACHE STRING
+    "Choose the type of build, options are: Debug, Release, RelWithDebInfo and MinSizeRel."
+    FORCE)
```

Comment:
Why CACHE?  Why the description?  Why FORCE?

Reply:
I need to override the default CMake variable if I want to set a default build type on my own because I touch a `CACHE` variable. 
Omitting `CACHE` creates a local variable instead and `CACHE` variables require a docstring. If I omit `FORCE` and the `CACHE` variable already exists, nothing happens. Of course, I only want to force a build type if it is empty. Hence, the check above. Also, compare https://blog.kitware.com/cmake-and-the-default-build-type/.

Reply:
I need to override the default CMake variable if I want to set a default build type on my own because I touch a CACHE variable.
Omitting CACHE creates a local variable instead and CACHE variables require a docstring. If I omit FORCE and the CACHE variable already exists, nothing happens. Of course, I only want to force a build type if it is empty. Hence, the check above. Also, compare https://blog.kitware.com/cmake-and-the-default-build-type/.

Reply:
@dalg24 Does @masterleinad's reply answer your question? If yes, please approve the PR.